### PR TITLE
community[minor]: Postgres Integration for Chat Memory

### LINKED ISSUE #4123: Postgres Integration for Chat Memory
It would be helpful to have postgres integration like what langchain.py currently has: [Link](https://python.langchain.com/docs/integrations/memory/postgres_chat_message_history)

### PR DESCRIPTION
**Pull request by MJDeligan**
Closes #4123 .

Chat message history backed by Postgres. Implementation based off of the [Python version](https://github.com/langchain-ai/langchain/blob/master/libs/community/langchain_community/chat_message_histories/postgres.py) and other memory store integration in langchainjs.

The constructor has an optional flag to escape the table name as suggested for other postgres integrations by @jacoblee93  [here](https://github.com/langchain-ai/langchainjs/pull/4398#issuecomment-1945070785).

Includes integration tests from oter

### CODE REVIEW

File: libs/langchain-community/src/stores/message/postgres.ts [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    // Ensure that either a client or config is provided
+    if (!pool && !poolConfig) {
+      throw new Error(
+        "PostgresChatMessageHistory requires either a pool instance or pool config"
+      );
+    }
+    this.pool = pool ?? new pg.Pool(poolConfig);
+    const _tableName = tableName || this.tableName;
+    this.tableName = escapeTableName
+      ? pg.escapeIdentifier(_tableName)
```
Comment (jacoblee93) [inline libs/langchain-community/src/stores/message/postgres.ts:102]: I am getting a build error here - I think it's missing from the types?

**Comment by jacoblee93** [comment]
Looks good except for one potential issue with types!

**Comment by MJDeligan** [comment]
Pushed a small fix to the example. Works great! 

The types for pg were outdated, Bumping them fixes the build. Thanks for the help @jacoblee93

**Comment by jacoblee93** [comment]
Ah, thank you!

**Pull request merged**